src/transformers/models/bart/modeling_tf_bart.py

```diff
@@ -886,35 +884,35 @@ def call(
         # embed positions
         positions = self.embed_positions(input_shape, past_key_values_length)
 
-        if inputs_embeds is None:
-            inputs_embeds = self.embed_tokens(inputs["input_ids"])
-        else:
-            inputs_embeds = inputs["inputs_embeds"]
+        if inputs["inputs_embeds"] is None:
+            inputs["inputs_embeds"] = self.embed_tokens(inputs["input_ids"])
 
-        hidden_states = inputs_embeds * self.embed_scale
+        hidden_states = inputs["inputs_embeds"] * self.embed_scale
 
         # [bsz, seq_len] -> [bsz, 1, tgt_seq_len, src_seq_len]
-        combined_attention_mask = None
+        combined_attention_mask = tf.fill(dims=input_shape, value=-1.0)
         if input_shape[-1] > 1:
             combined_attention_mask = _make_causal_mask(input_shape, past_key_values_length=past_key_values_length)
 
+        is_combined_None = tf.reduce_any(combined_attention_mask == -1)
+
         if inputs["attention_mask"] is None and inputs["input_ids"] is not None and input_shape[-1] > 1:
-            attention_mask = tf.cast(
+            inputs["attention_mask"] = tf.cast(
                 tf.math.not_equal(inputs["input_ids"], self.config.pad_token_id), inputs["input_ids"].dtype
             )
         else:
-            attention_mask = tf.ones(input_shape, dtype=tf.int32)
+            inputs["attention_mask"] = tf.ones(input_shape, dtype=tf.int32)
 
-        if attention_mask is not None and combined_attention_mask is not None:
-            # [bsz, seq_len] -> [bsz, 1, tgt_seq_len, src_seq_len]
-            combined_attention_mask = combined_attention_mask + _expand_mask(
-                attention_mask, past_key_values_length=past_key_values_length
+        def combined_false():
+            return combined_attention_mask + _expand_mask(
+                inputs["attention_mask"], past_key_values_length=past_key_values_length
             )
 
-        encoder_hidden_states = inputs["encoder_hidden_states"]
-        if encoder_hidden_states is not None and inputs["encoder_attention_mask"] is not None:
+        combined_attention_mask = tf.cond(is_combined_None, lambda: combined_attention_mask, combined_false)
```

Comment:
This is starting to be a bit messy. Can we create a function `make_combined_mask` and a function `make_fake_combined_mask` to simplify all of this?

- As state above, `is_combined_None` can simply be `input_shape[-1] > 1`.
- `make_fake_combined_mask` would return `tf.fill(dims=input_shape, value=-1.0)`
- `make_combined_mask` would do the `_make_causal_mask` then add the `_expand_mask`

Does that make sense?

Reply:
It makes sense 👍